tree/ntuple/v7/inc/ROOT/RNTupleView.hxx

```diff
@@ -185,37 +174,37 @@ public:
 
    RNTupleGlobalRange GetFieldRange() const { return RNTupleGlobalRange(0, fField.GetNElements()); }
 
-   template <typename C = T>
-   typename std::enable_if_t<Internal::IsMappable<FieldT>::value, const C&>
-   operator()(NTupleSize_t globalIndex) { return *fField.Map(globalIndex); }
-
-   template <typename C = T>
-   typename std::enable_if_t<!Internal::IsMappable<FieldT>::value, const C&>
-   operator()(NTupleSize_t globalIndex) {
-      fField.Read(globalIndex, &fValue);
-      return *fValue.Get<T>();
+   const T &operator()(NTupleSize_t globalIndex)
+   {
+      if constexpr (Internal::isMappable<FieldT>)
+         return *fField.Map(globalIndex);
+      else {
+         fField.Read(globalIndex, &fValue);
+         return *fValue.Get<T>();
+      }
    }
 
-   template <typename C = T>
-   typename std::enable_if_t<Internal::IsMappable<FieldT>::value, const C&>
-   operator()(const RClusterIndex &clusterIndex) { return *fField.Map(clusterIndex); }
-
-   template <typename C = T>
-   typename std::enable_if_t<!Internal::IsMappable<FieldT>::value, const C&>
-   operator()(const RClusterIndex &clusterIndex) {
-      fField.Read(clusterIndex, &fValue);
-      return *fValue.Get<T>();
+   const T &operator()(const RClusterIndex &clusterIndex)
+   {
+      if constexpr (Internal::isMappable<FieldT>)
+         return *fField.Map(clusterIndex);
+      else {
+         fField.Read(clusterIndex, &fValue);
+         return *fValue.Get<T>();
+      }
    }
 
-   template <typename C = T>
-   typename std::enable_if_t<Internal::IsMappable<FieldT>::value, const C*>
-   MapV(NTupleSize_t globalIndex, NTupleSize_t &nItems) {
+   // TODO(bgruber): turn enable_if into requires clause with C++20
+   template <typename C = T, std::enable_if_t<Internal::isMappable<FieldT>, C*> = nullptr>
+   const C *MapV(NTupleSize_t globalIndex, NTupleSize_t &nItems)
```

Comment:
This change and the next hunk are more a matter of style. I consider it more readable to see the return type and move the `enable_if` into the template signature.